controller/deployments.go

```diff
@@ -3,25 +3,31 @@ package controller
 import (
 	"context"
 	"encoding/json"
-	"io/ioutil"
-	"net/http"
 	"net/url"
 	"os"
 	"time"
 
 	"github.com/fabric8-services/fabric8-wit/app"
-	"github.com/fabric8-services/fabric8-wit/auth"
-	"github.com/fabric8-services/fabric8-wit/auth/authservice"
 	"github.com/fabric8-services/fabric8-wit/configuration"
 	"github.com/fabric8-services/fabric8-wit/errors"
 	"github.com/fabric8-services/fabric8-wit/jsonapi"
 	"github.com/fabric8-services/fabric8-wit/kubernetes"
 	"github.com/fabric8-services/fabric8-wit/log"
+
 	"github.com/goadesign/goa"
 	errs "github.com/pkg/errors"
 	uuid "github.com/satori/go.uuid"
 )
 
+/*
+ The deployments API can operate directly talking to Auth and OSO,
+ or it can use the fabric8-tenant service as a replacement for fabric8-auth
+ and the fabric8-froxy service as a replacement for direct OpenShift API calls.
+
+ If the tenant service is available, it will be used in place of auth.
+ If the proxy service is available, it will be used in place of direct calls.
```

Comment:
Why do we need to support both targets ? (tenant service vs auth, and proxy service vs direct calls to OSO)

Reply:
Actually, this comment is out of date.  I'll remove it since it serves no purpose in the code at that point.

The current code uses the WIT tenant endpoint to talk to Auth, and can operate with or without a proxy server.  In fact, it uses the proxy service (if present) to talk to the k8s API, and must make direct calls into k8s for metrics, console and logs - because the proxy doesn't yet handle those.

Reply:
ok, so may I ask why you want to pass through `tenant` to talk to `auth` ? the `auth` package already provides a function to get a client to the `auth` service.

Reply:
ok, great! thanks for the clarification!